tsdb/db_test.go

```diff
@@ -889,10 +887,14 @@ func TestWALSegmentSizeOptions(t *testing.T) {
 			db, closeFn := openTestDB(t, opts, nil)
 			defer closeFn()
 
-			app := db.Appender()
 			for i := int64(0); i < 155; i++ {
-				_, err := app.Add(labels.Labels{labels.Label{Name: "wal", Value: "size"}}, i, rand.Float64())
+				app := db.Appender()
+				ref, err := app.Add(labels.Labels{labels.Label{Name: "wal" + fmt.Sprintf("%d", i), Value: "size"}}, i, rand.Float64())
```

Comment:
Is this related to this PR? :thinking:

Reply:
Yes, as explained in or comments, when we reused a committed appender we accidentally logged all already committed samples again and again. With the new PR we need to append more samples to log() enough and get to the same wal size as before.